BuildResidentialHPXML/measure.rb

```diff
@@ -5755,8 +5809,17 @@ def self.set_pv_systems(hpxml, args, epw_file)
   def self.set_battery(hpxml, args)
     return unless args[:battery_present]
 
+    foundation_type = hpxml.foundations[-1].foundation_type
+    attic_type = hpxml.attics[-1].attic_type
+
     if args[:battery_location].is_initialized
       location = args[:battery_location].get
+
+      if location == HPXML::LocationCrawlspace
+        location = get_location_from_foundation_type(location, foundation_type)
+      elsif location == HPXML::LocationAttic
+        location = get_location_from_attic_type(location, attic_type)
+      end
```

Comment:
Instead of duplicating this logic three times, just merge the two `get_location_from_foo` methods into a single `get_location` and move this logic into that method. Then all you need in these three places is e.g. `location = get_location(args[:foo_location]).get`.